### ISSUE TITLE
Please include deleting ASP.NET Core Runtime in Uninstaller

### ISSUE THREAD

**Issue by JunTaoLuo** [aspnet/Home#3286]
_From @JunTaoLuo on June 29, 2018 21:42_

_From @VSDekar on June 29, 2018 12:2_

We had huge problems with dotnet core SDK 2.1.301 so we went back to 2.1.300. But the uninstaller did not delete the ASP.NET Core runtime.

Because the runtime was still there after we uninstalled 2.1.301 we got some very strange problems.

So my expectation is, that everything that an installer installs on the system will be deleted by the uninstaller. I had to delete manually the folder in `C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App`

Thx

_Copied from original issue: dotnet/core#1740_

_Copied from original issue: aspnet/SetupSharedFramework#10_

**Comment by JunTaoLuo** [comment]
_From @leecow on June 29, 2018 15:44_

@JunTaoLuo - can you have a look and move this issue to the right repo?

**Comment by JunTaoLuo** [comment]
@joeloff can you investigate why uninstalling the runtime did not remove the files under Microsoft.AspNetCore.App?

**Comment by JunTaoLuo** [comment]
_From @joeloff on June 29, 2018 22:6_

What exactly is installed on the machine? The ASP.NET shared framework ships in multiple ways

1. The .NET Core SDK
2. By itself as a standalone EXE.
3. Windows hosting bundle

For 1 & 2, the same MSI would be ref counted against all bundles (EXEs) to avoid removing it if another bundle is still installed.

The setup logs should be able to shed some light on why the MSI was skipped. Also, if the CLI marks the MSI as a permanent component, it won't remove it.

**Comment by JunTaoLuo** [comment]
Actually, although this is the most appropriate repo for this discussion, I should probably move it to Home as that's a public repo.

**Comment by VSDekar** [comment]
@JunTaoLuo
@joeloff 

We followed the Links from the blog post [.NET Core 2.1 June Update](https://blogs.msdn.microsoft.com/dotnet/2018/06/22/net-core-2-1-june-update/) and installed the SDK 64bit `https://download.microsoft.com/download/D/0/4/D04C5489-278D-4C11-9BD3-6128472A7626/dotnet-sdk-2.1.301-win-x64.exe` as far as I know.

To remove it: Control Panel --> Programs and Features --> Microsoft .NET Core SDK 2.1.301 --->  Uninstall

Do you need anything else? Where are the logs located? Just tell me what I can do to help you. (We had around 10 Devs with the very same problem) Is there a proper way aka uninstaller to "delete" the mentioned folder? I ask because I did not found a dedicated uninstaller for this. Since it is a working computer I can provide infos on Monday.

**Comment by joeloff** [comment]
The logs will be in your %temp% folder. The files will all start with something similar to Microsoft_.NET_Core_SDK_2.1.400_-_preview_(x86)_20180629212714.log.

You could delete the folder, but internally the OS will still track them as being installed.

Hopefully I can provide a better solution once I examine the log files

**Comment by VSDekar** [comment]
Ok i have found the logs.
5 Log files:

- Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500.log 15KB
- Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_000_dotnet_sdk_internal_2.1.301_win_x64.msi.log 2759KB
- Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_001_dotnet_host_2.1.1_win_x64.msi.log 88KB
- Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_002_dotnet_hostfxr_2.1.1_win_x64.msi.log 88KB
- Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_003_dotnet_runtime_2.1.1_win_x64.msi.log 438KB

Here is the first log file:
```
[22BC:0E40][2018-06-29T13:45:00]i001: Burn v3.10.4.4718, Windows v10.0 (Build 17134: Service Pack 0), path: C:\ProgramData\Package Cache\{faf6feb2-1788-41aa-ae3b-4aed0ec403d7}\dotnet-sdk-2.1.301-win-x64.exe
[22BC:0E40][2018-06-29T13:45:00]i000: Initializing string variable 'DOTNETHOME' to value '[ProgramFiles6432Folder]dotnet'
[22BC:0E40][2018-06-29T13:45:00]i000: Initializing string variable 'BUNDLEMONIKER' to value 'Microsoft .NET Core SDK 2.1.301'
[22BC:0E40][2018-06-29T13:45:00]i000: Initializing string variable 'DOTNETSDKVERSION' to value '2.1.301'
[22BC:0E40][2018-06-29T13:45:00]i000: Initializing string variable 'DOTNETRUNTIMEVERSION' to value '2.1.1'
[22BC:0E40][2018-06-29T13:45:00]i000: Initializing string variable 'ASPNETCOREVERSION' to value '2.1.1'
[22BC:0E40][2018-06-29T13:45:00]i009: Command Line: '"-burn.clean.room=C:\ProgramData\Package Cache\{faf6feb2-1788-41aa-ae3b-4aed0ec403d7}\dotnet-sdk-2.1.301-win-x64.exe" -burn.filehandle.attached=588 -burn.filehandle.self=600 /uninstall'
[22BC:0E40][2018-06-29T13:45:00]i000: Setting string variable 'WixBundleLog' to value 'C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500.log'
[22BC:0E40][2018-06-29T13:45:00]i000: Setting string variable 'WixBundleManufacturer' to value 'Microsoft Corporation'
[22BC:0D38][2018-06-29T13:45:00]i000: Setting numeric variable 'WixStdBALanguageId' to value 1033
[22BC:0D38][2018-06-29T13:45:00]i000: Setting version variable 'WixBundleFileVersion' to value '2.1.301.0'
[22BC:0E40][2018-06-29T13:45:00]i100: Detect begin, 5 packages
[22BC:0E40][2018-06-29T13:45:00]i101: Detected package: dotnet_runtime_2.1.1_win_x64.msi, state: Present, cached: Complete
[22BC:0E40][2018-06-29T13:45:00]i101: Detected package: dotnet_hostfxr_2.1.1_win_x64.msi, state: Present, cached: Complete
[22BC:0E40][2018-06-29T13:45:00]i101: Detected package: dotnet_host_2.1.1_win_x64.msi, state: Present, cached: Complete
[22BC:0E40][2018-06-29T13:45:00]i101: Detected package: dotnet_sdk_internal_2.1.301_win_x64.msi, state: Present, cached: Complete
[22BC:0E40][2018-06-29T13:45:00]i101: Detected package: AspNetCoreSharedFramework_x64, state: Present, cached: Complete
[22BC:0E40][2018-06-29T13:45:00]i199: Detect complete, result: 0x0
[22BC:0E40][2018-06-29T13:45:23]i200: Plan begin, 5 packages, action: Uninstall
[22BC:0E40][2018-06-29T13:45:23]w327: Will not uninstall package: AspNetCoreSharedFramework_x64, found dependents: 1
[22BC:0E40][2018-06-29T13:45:23]w328: Found dependent: {f257de86-106a-419b-8783-e0539f790906}, name: Microsoft .NET Core SDK 2.1.300 (x64)
[22BC:0E40][2018-06-29T13:45:23]i000: Setting string variable 'WixBundleRollbackLog_dotnet_sdk_internal_2.1.301_win_x64.msi' to value 'C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_000_dotnet_sdk_internal_2.1.301_win_x64.msi_rollback.log'
[22BC:0E40][2018-06-29T13:45:23]i000: Setting string variable 'WixBundleLog_dotnet_sdk_internal_2.1.301_win_x64.msi' to value 'C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_000_dotnet_sdk_internal_2.1.301_win_x64.msi.log'
[22BC:0E40][2018-06-29T13:45:23]i000: Setting string variable 'WixBundleRollbackLog_dotnet_host_2.1.1_win_x64.msi' to value 'C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_001_dotnet_host_2.1.1_win_x64.msi_rollback.log'
[22BC:0E40][2018-06-29T13:45:23]i000: Setting string variable 'WixBundleLog_dotnet_host_2.1.1_win_x64.msi' to value 'C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_001_dotnet_host_2.1.1_win_x64.msi.log'
[22BC:0E40][2018-06-29T13:45:23]i000: Setting string variable 'WixBundleRollbackLog_dotnet_hostfxr_2.1.1_win_x64.msi' to value 'C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_002_dotnet_hostfxr_2.1.1_win_x64.msi_rollback.log'
[22BC:0E40][2018-06-29T13:45:23]i000: Setting string variable 'WixBundleLog_dotnet_hostfxr_2.1.1_win_x64.msi' to value 'C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_002_dotnet_hostfxr_2.1.1_win_x64.msi.log'
[22BC:0E40][2018-06-29T13:45:23]i000: Setting string variable 'WixBundleRollbackLog_dotnet_runtime_2.1.1_win_x64.msi' to value 'C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_003_dotnet_runtime_2.1.1_win_x64.msi_rollback.log'
[22BC:0E40][2018-06-29T13:45:23]i000: Setting string variable 'WixBundleLog_dotnet_runtime_2.1.1_win_x64.msi' to value 'C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_003_dotnet_runtime_2.1.1_win_x64.msi.log'
[22BC:0E40][2018-06-29T13:45:23]i201: Planned package: AspNetCoreSharedFramework_x64, state: Present, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: No, dependency: Unregister
[22BC:0E40][2018-06-29T13:45:23]i201: Planned package: dotnet_sdk_internal_2.1.301_win_x64.msi, state: Present, default requested: Absent, ba requested: Absent, execute: Uninstall, rollback: Install, cache: No, uncache: Yes, dependency: Unregister
[22BC:0E40][2018-06-29T13:45:23]i201: Planned package: dotnet_host_2.1.1_win_x64.msi, state: Present, default requested: Absent, ba requested: Absent, execute: Uninstall, rollback: Install, cache: No, uncache: Yes, dependency: Unregister
[22BC:0E40][2018-06-29T13:45:23]i201: Planned package: dotnet_hostfxr_2.1.1_win_x64.msi, state: Present, default requested: Absent, ba requested: Absent, execute: Uninstall, rollback: Install, cache: No, uncache: Yes, dependency: Unregister
[22BC:0E40][2018-06-29T13:45:23]i201: Planned package: dotnet_runtime_2.1.1_win_x64.msi, state: Present, default requested: Absent, ba requested: Absent, execute: Uninstall, rollback: Install, cache: No, uncache: Yes, dependency: Unregister
[22BC:0E40][2018-06-29T13:45:23]i299: Plan complete, result: 0x0
[22BC:0E40][2018-06-29T13:45:23]i300: Apply begin
[22BC:0E40][2018-06-29T13:45:23]i010: Launching elevated engine process.
[22BC:0E40][2018-06-29T13:45:23]i011: Launched elevated engine process.
[22BC:0E40][2018-06-29T13:45:23]i012: Connected to elevated engine.
[3E9C:3EA8][2018-06-29T13:45:23]i358: Pausing automatic updates.
[3E9C:3EA8][2018-06-29T13:45:23]i359: Paused automatic updates.
[3E9C:3EA8][2018-06-29T13:45:23]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{faf6feb2-1788-41aa-ae3b-4aed0ec403d7}, resume: Active, restart initiated: No, disable resume: No
[3E9C:3EA8][2018-06-29T13:45:23]i326: Removed dependency: {faf6feb2-1788-41aa-ae3b-4aed0ec403d7} on package provider: Microsoft.AspNetCore.SharedFramework_x64_ENU,v2.1,rtm, package AspNetCoreSharedFramework_x64
[3E9C:3EA8][2018-06-29T13:45:23]i326: Removed dependency: {faf6feb2-1788-41aa-ae3b-4aed0ec403d7} on package provider: Dotnet_CLI_2.1.301_x64, package dotnet_sdk_internal_2.1.301_win_x64.msi
[3E9C:3EA8][2018-06-29T13:45:23]i301: Applying execute package: dotnet_sdk_internal_2.1.301_win_x64.msi, action: Uninstall, path: (null), arguments: ' ARPSYSTEMCOMPONENT="1" DOTNETHOME="C:\Program Files\dotnet" EXEFULLPATH="C:\Users\<user>\Downloads\dotnet-sdk-2.1.301-win-x64 (1).exe"'
[22BC:0E40][2018-06-29T13:45:35]i319: Applied execute package: dotnet_sdk_internal_2.1.301_win_x64.msi, result: 0x0, restart: None
[3E9C:3EA8][2018-06-29T13:45:35]i326: Removed dependency: {faf6feb2-1788-41aa-ae3b-4aed0ec403d7} on package provider: Dotnet_CLI_SharedHost_16.68.26606_x64, package dotnet_host_2.1.1_win_x64.msi
[3E9C:3EA8][2018-06-29T13:45:35]i301: Applying execute package: dotnet_host_2.1.1_win_x64.msi, action: Uninstall, path: (null), arguments: ' ARPSYSTEMCOMPONENT="1" DOTNETHOME="C:\Program Files\dotnet"'
[22BC:0E40][2018-06-29T13:45:36]i319: Applied execute package: dotnet_host_2.1.1_win_x64.msi, result: 0x0, restart: None
[3E9C:3EA8][2018-06-29T13:45:36]i326: Removed dependency: {faf6feb2-1788-41aa-ae3b-4aed0ec403d7} on package provider: Dotnet_CLI_HostFxr_2.1.1_x64, package dotnet_hostfxr_2.1.1_win_x64.msi
[3E9C:3EA8][2018-06-29T13:45:36]i301: Applying execute package: dotnet_hostfxr_2.1.1_win_x64.msi, action: Uninstall, path: (null), arguments: ' ARPSYSTEMCOMPONENT="1" DOTNETHOME="C:\Program Files\dotnet"'
[22BC:0E40][2018-06-29T13:45:37]i319: Applied execute package: dotnet_hostfxr_2.1.1_win_x64.msi, result: 0x0, restart: None
[3E9C:3EA8][2018-06-29T13:45:37]i326: Removed dependency: {faf6feb2-1788-41aa-ae3b-4aed0ec403d7} on package provider: DotNet.CLI.SharedFramework.Microsoft.NETCore.App_2.1.1_x64, package dotnet_runtime_2.1.1_win_x64.msi
[3E9C:3EA8][2018-06-29T13:45:37]i301: Applying execute package: dotnet_runtime_2.1.1_win_x64.msi, action: Uninstall, path: (null), arguments: ' ARPSYSTEMCOMPONENT="1" DOTNETHOME="C:\Program Files\dotnet"'
[22BC:0E40][2018-06-29T13:45:39]i319: Applied execute package: dotnet_runtime_2.1.1_win_x64.msi, result: 0x0, restart: None
[3E9C:3EA8][2018-06-29T13:45:39]i351: Removing cached package: dotnet_sdk_internal_2.1.301_win_x64.msi, from path: C:\ProgramData\Package Cache\{11E5A48E-42CA-4E73-BE16-19466DF32F2B}v8.91.25290\
[3E9C:3EA8][2018-06-29T13:45:39]i351: Removing cached package: dotnet_host_2.1.1_win_x64.msi, from path: C:\ProgramData\Package Cache\{009F24CC-E2B4-4E04-A38E-7210273FD2E2}v16.68.26606\
[3E9C:3EA8][2018-06-29T13:45:39]i351: Removing cached package: dotnet_hostfxr_2.1.1_win_x64.msi, from path: C:\ProgramData\Package Cache\{91AB97C2-FA5F-42EB-AB4F-92AADCFF928E}v16.68.26606\
[3E9C:3EA8][2018-06-29T13:45:39]i351: Removing cached package: dotnet_runtime_2.1.1_win_x64.msi, from path: C:\ProgramData\Package Cache\{F95EA188-25DF-4AE7-BDAA-4184BCBB8D9A}v16.68.26606\
[3E9C:3EA8][2018-06-29T13:45:39]i372: Session end, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{faf6feb2-1788-41aa-ae3b-4aed0ec403d7}, resume: None, restart: None, disable resume: No
[3E9C:3EA8][2018-06-29T13:45:39]i330: Removed bundle dependency provider: {faf6feb2-1788-41aa-ae3b-4aed0ec403d7}
[3E9C:3EA8][2018-06-29T13:45:39]i352: Removing cached bundle: {faf6feb2-1788-41aa-ae3b-4aed0ec403d7}, from path: C:\ProgramData\Package Cache\{faf6feb2-1788-41aa-ae3b-4aed0ec403d7}\
[3E9C:3EA8][2018-06-29T13:45:39]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{faf6feb2-1788-41aa-ae3b-4aed0ec403d7}, resume: None, restart initiated: No, disable resume: No
[22BC:0E40][2018-06-29T13:45:39]i399: Apply complete, result: 0x0, restart: None, ba requested restart:  No
[22BC:0E40][2018-06-29T13:49:27]i500: Shutting down, exit code: 0x0
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: ASPNETCOREVERSION = 2.1.1
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: BUNDLEMONIKER = Microsoft .NET Core SDK 2.1.301
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: DOTNETHOME = C:\Program Files\dotnet
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: DOTNETRUNTIMEVERSION = 2.1.1
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: DOTNETSDKVERSION = 2.1.301
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: ProgramFiles6432Folder = C:\Program Files\
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleAction = 3
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleElevated = 1
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleFileVersion = 2.1.301.0
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleInstalled = 1
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleLog = C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500.log
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleLog_dotnet_host_2.1.1_win_x64.msi = C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_001_dotnet_host_2.1.1_win_x64.msi.log
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleLog_dotnet_hostfxr_2.1.1_win_x64.msi = C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_002_dotnet_hostfxr_2.1.1_win_x64.msi.log
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleLog_dotnet_runtime_2.1.1_win_x64.msi = C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_003_dotnet_runtime_2.1.1_win_x64.msi.log
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleLog_dotnet_sdk_internal_2.1.301_win_x64.msi = C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_000_dotnet_sdk_internal_2.1.301_win_x64.msi.log
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleManufacturer = Microsoft Corporation
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleName = Microsoft .NET Core SDK 2.1.301 (x64)
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleOriginalSource = C:\Users\<user>\Downloads\dotnet-sdk-2.1.301-win-x64 (1).exe
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleOriginalSourceFolder = C:\Users\<user>\Downloads\
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleProviderKey = {faf6feb2-1788-41aa-ae3b-4aed0ec403d7}
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleRollbackLog_dotnet_host_2.1.1_win_x64.msi = C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_001_dotnet_host_2.1.1_win_x64.msi_rollback.log
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleRollbackLog_dotnet_hostfxr_2.1.1_win_x64.msi = C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_002_dotnet_hostfxr_2.1.1_win_x64.msi_rollback.log
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleRollbackLog_dotnet_runtime_2.1.1_win_x64.msi = C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_003_dotnet_runtime_2.1.1_win_x64.msi_rollback.log
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleRollbackLog_dotnet_sdk_internal_2.1.301_win_x64.msi = C:\Users\<user>\AppData\Local\Temp\Microsoft_.NET_Core_SDK_2.1.301_(x64)_20180629134500_000_dotnet_sdk_internal_2.1.301_win_x64.msi_rollback.log
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleSourceProcessFolder = C:\ProgramData\Package Cache\{faf6feb2-1788-41aa-ae3b-4aed0ec403d7}\
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleSourceProcessPath = C:\ProgramData\Package Cache\{faf6feb2-1788-41aa-ae3b-4aed0ec403d7}\dotnet-sdk-2.1.301-win-x64.exe
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleTag = 
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixBundleVersion = 2.1.301.0
[22BC:0E40][2018-06-29T13:49:27]i410: Variable: WixStdBALanguageId = 1033
[22BC:0E40][2018-06-29T13:49:27]i007: Exit code: 0x0, restarting: No
```

I think the answer to my "problem" here are these lines:
```
[22BC:0E40][2018-06-29T13:45:23]i200: Plan begin, 5 packages, action: Uninstall
[22BC:0E40][2018-06-29T13:45:23]w327: Will not uninstall package: AspNetCoreSharedFramework_x64, found dependents: 1
[22BC:0E40][2018-06-29T13:45:23]w328: Found dependent: {f257de86-106a-419b-8783-e0539f790906}, name: Microsoft .NET Core SDK 2.1.300 (x64)
```

So the question is. Why the AspNetCoreSharedFramework_x64 has a dependency on the 2.1.300 SDK? This should not be the case i think? Why has the AspNetCoreSharedFramework_x64 no version number?

Anyway, hope you can work with that. If you need the other log files, let me know where i can upload them.

**Comment by joeloff** [comment]
The .NET Core SDKs install SxS. If two bundles (EXE installers) ship the same copy of an MSI, the MSI is reference counted and won't be removed until all the EXEs that reference it are removed. For the uninstall you ran it would simply decrement the reference.

The shared framework does have a version, but the bundle log will typically only report the internal package ID when logging things like bundle dependencies.

I'll take a look and see whether the two copies are in fact the same version of the framework. I would have expected that the two SDKs shipped different version of it.

**Comment by VSDekar** [comment]
Somewhere something must be wired the wrong way. Because of the folder within `C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App` which gets not deleted had the name `2.1.1` which I think is part of the 2.1.301 SDK installation.

**Comment by VSDekar** [comment]
@joeloff Thx for the cmd.

I am amazed that you take so much effort just because a random guy appeared and said something does not get uninstalled! Good job! I really appreciate that! 🥇

**Comment by joeloff** [comment]
You're welcome.

**Comment by Eilon** [comment]
@joeloff - I've assigned to you so that you can close once the bug is fixed in the private repo.

**Comment by joeloff** [comment]
The fix was merged into the repo last week for 2.1.3.
